Cockpit 244 (#409)
cockpit-project/cockpit-project.github.io: _posts/2021-05-12-244.md

---
title: Cockpit 244
author: kkoukiou
date: 2021-05-12
tags: cockpit linux
slug: cockpit-244
category: release
summary: Cockpit with various improvements
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 244 and Cockpit-Machines 244.

## Shell: No opportunistic sudo invocation

> Cockpit will no longer try to run sudo on the very first login, and you have to
> explicitly turn on administrative access once for each browser profile that you
> use to access Cockpit. This prevents cockpit's use of sudo being noisy in the
> journal.

Reviewer: ```suggestion
Cockpit will no longer try to automatically run sudo on the very first login: you'll have to
explicitly turn on administrative access once for each browser profile that you
use to access Cockpit. This reduces a lot of noise about authentication failures in the journal.
```

> ![Sudo error messages](/images/sudo-error-messages.png)

Reviewer 2: It seems to me that the screenshot does not match the description. The PR has this additional paragraph: "In addition, Cockpit now shows sudo's exact messages in the "Administrative access" dialog.", which is the explanation for the screenshot. Can you please add that?

Author: Ups right, sorry.

Reviewer 3: Oh, I was assuming we wouldn't have a acreenshot. But if we would, I agree it should be something else. Although this one has strong https://xkcd.com/838/ vibes...

![incident](https://user-images.githubusercontent.com/10246/117935594-2e1fa900-b304-11eb-9b4e-10d8b2c9d58d.png)

## Machines: Edit the MAC address of a VM's network interface

> To edit the MAC address of a virtual machine click the `Edit` button on the row

Reviewer: nit: I wonder if backticks are the right markup option here?  Is there a better tag for "UI element" that doesn't imply "fixed width"?

Author: No idea, to be honest we never did guide the user int the UI in this way in the past in a blog post. I was not even sure if we want this kind of Ui guidance here.

Reviewer: Ya.  I agree.  it's oddly specific.  Maybe take it out?

Reviewer 3: It could be "Edit", as we're quoting the button's text. This is pretty standard in software documentation.

```suggestion
To edit the MAC address of a virtual machine click the "Edit" button on the row
```

There's also the `<kbd>` tag, but that's more for keystrokes, not UI elements. And it also makes the text fixed-width.

Author: I have no problem @allisonkarlitskaya, I just added as a filler to be honest. @garrett shall I take it out?

of the network interface. A common scenario where this is needed is when
configuring the DHCP server to map a static IP address to a certain MAC address.

![VM edit MAC address](/images/vm-edit-mac-address.png)

### Try it out

Cockpit 244 is available now:

 * [For your Linux system](https://cockpit-project.org/running.html)
 * [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/244)

>  * [Cockpit Fedora 33]()

Reviewer: bodhi URLs of course...

---
title: Cockpit 244
author: kkoukiou
date: 2021-05-12
tags: cockpit linux
slug: cockpit-244
category: release
summary: Cockpit with various improvements
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

> Here are the release notes from Cockpit version 244 and Cockpit-Machines 244.

Reviewer: "Cockpit-Machines" looks kinda funny...  Is that actually how we write it?

Author: I can also change it to Cockpit Machines.

Author: We haven't standardized how we refer to it actually. Same for podman.

Author: Removed the hiphen.

> ## Shell: No opportunistic sudo invocation

Reviewer 3: I feel this is a strange headline:

- It's weird to have this be a negative with "no"
- "opportunistic" is odd
- "invocation" is also odd (unless you're in a church)

Author: Ok this comment is super funny, but does not help me much.

When you 'call' sudo, or any command it's invocation. It's a common word to use I feel.

Slightly knowing what that change is about, I think this is actually a short but descriptive header.

Reviewer 3: https://en.wikipedia.org/wiki/Invocation:

> An invocation (from the Latin verb invocare "to call on, invoke, to give") may take the form of:
> - Supplication, prayer or spell.
> - A form of possession.
> - Command or conjuration.
> - Self-identification with certain spirits.

Author: Hmm let me quickly edit wikipedia to add how else it can be used :)

https://linux.die.net/man/8/sudo if you search for `invok` there are 11 results on the sudo man page.

Reviewer 3: If you want to use a form of the word that isn't so supernatural:

```suggestion
## Shell: Sudo is no longer invoked unless needed
```

But that's still a bit problematic.

```suggestion
## Shell: Sudo is used only when necessary
```

I don't know. What do you think?

Reviewer 3: But that last one makes people think "was sudo used willy-nilly?" (that is for no good reason whatsoever, haphazardly, in a sloppy way, etc.)

Author: 'Unless needed' or 'when necessary' implies that it will be always invoked automatically when needed, but it's not the case. The user needs to turn on the 'Admin access'

Reviewer 4: What about "Shell: sudo will only be invoked when explicitly requested"

Author: ok!

Reviewer 3: Sure, that works for me.

Reviewer 3: Oh wait. Minor revision:

Shell: sudo is invoked only when explicitly requested

Author: Fixed.

Cockpit will no longer try to automatically run sudo on the very first login: you'll have to
explicitly turn on administrative access once for each browser profile that you
use to access Cockpit. This reduces a lot of noise about authentication failures in the journal.

In addition, Cockpit now shows sudo's exact messages in the "Administrative access" dialog.

![Sudo error messages](/images/sudo-error-messages.png)

## Machines: Edit the MAC address of a VM's network interface

To edit the MAC address of a virtual machine click the `Edit` button on the row
of the network interface. A common scenario where this is needed is when
configuring the DHCP server to map a static IP address to a certain MAC address.

> ![VM edit MAC address](/images/vm-edit-mac-address.png)

Reviewer 3: I didn't see this PR fly by. (I don't remember seeing it when I checked "release notes" tags on Monday...)

Sadly, the screenshot clearly shows improper alignment of the MAC address label. :cry: 

![Screenshot from 2021-05-12 09-29-29](https://user-images.githubusercontent.com/10246/117935993-9ec6c580-b304-11eb-93ab-c4cb1d735027.png)

This should be fixed... in a follow-up, I guess...

Reviewer 3: (The two strings should be aligned to the baseline. Sure, the other form elements still have a bug where it's off by 1 or 2 pixels, based on rounding.... but that's an upstream PF issue. But this is _wildly_ off.)

Author: OH it's because the PR had another screenshot which was obsolete, so I took a new one. 
Let me check if I can quickly fix it.

Author: https://github.com/cockpit-project/cockpit-machines/pull/158

Reviewer 3: Here's a new screenshot, cropped and optimized:
![nic-edit](https://user-images.githubusercontent.com/10246/117946412-945df900-b30f-11eb-9e3d-dd0c61a4c915.png)

Reviewer 3: (Unless you'd rather have me fix up the previous one. This one has fake MAC addresses though.)

Author: http://sqa.fyicenter.com/1000208_MAC_Address_Validator.html#Result
This mac address is not valid.

The screenshot I have is already cropped. How do you optimize it? Can I do it?

### Try it out

Cockpit 244 is available now:

 * [For your Linux system](https://cockpit-project.org/running.html)
 * [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/244)
 * [Cockpit Fedora 33](https://bodhi.fedoraproject.org/updates/FEDORA-2021-de41124d1b)

>  * [Cockpit Fedora 34](https://bodhi.fedoraproject.org/updates/FEDORA-2021-16d3cfc061)

Reviewer 2: As the announcement now speaks about c-machines as well, I think it should also mention the upstream/bodhi URLs for it?

Author: Done
